views.go

```diff
@@ -117,23 +117,83 @@ func ProcessViewResult(result ViewResult, params map[string]interface{},
 	return result, nil
 }
 
-func ReduceViewResult(reduceFunction string, result *ViewResult) error {
+func ReduceViewResult(reduceFunction string, params map[string]interface{}, result *ViewResult) error {
+	reduceFun, compileErr := ReduceFunc(reduceFunction)
+	if compileErr != nil {
+		return compileErr
+	}
+	groupLevel := 0
+	if params["group"] != nil && params["group"].(bool) == true {
+		groupLevel = -1
+	} else if params["group_level"] != nil {
+		groupLevel = int(params["group_level"].(uint64))
+	}
+	if groupLevel != 0 {
+		var collator JSONCollator
+		key := result.Rows[0].Key
+		if groupLevel != -1 {
+			// don't try to cast key as a slice if group=true
+			key = key.([]interface{})[0:groupLevel]
```

Comment:
I still don't understand this code at all .. what is it doing and why?  

how about key = doWeirdStuffToKey(groupLevel)

and name `doWeirdStuffToKey` to something that describes what it's doing, and some docs to help understand why, ideally with some before vs after examples.

(actually, after staring at this for a while I do now understand what it's going, but I think this can be refactored and documented to be a lot more clear about what's doing)